rsmtool/rsmcompare.py

```diff
@@ -180,44 +175,59 @@ def main():
     # set up the basic logging configuration
     formatter = LogFormatter()
 
-    handler = logging.StreamHandler(sys.stdout)
-    handler.setFormatter(formatter)
+    # we need two handlers, one that prints to stdout
+    # for the "run" command and one that prints to stderr
+    # from the "generate" command; the latter is necessary
+    # because do not want the warning to show up in the
+    # generated configuration file
+    stdout_handler = logging.StreamHandler(sys.stdout)
+    stdout_handler.setFormatter(formatter)
 
-    logging.root.addHandler(handler)
-    logging.root.setLevel(logging.INFO)
+    stderr_handler = logging.StreamHandler(sys.stderr)
+    stderr_handler.setFormatter(formatter)
 
-    # get a logger
+    logging.root.setLevel(logging.INFO)
     logger = logging.getLogger(__name__)
 
-    # set up an argument parser
-    parser = argparse.ArgumentParser(prog='rsmcompare')
-
-    parser.add_argument('config_file', help="The JSON configuration file for "
-                                            "this comparison")
+    # set up an argument parser via our helper function
+    parser = setup_rsmcmd_parser('rsmcompare',
+                                 uses_output_directory=True,
+                                 uses_subgroups=True)
+
+    # if the first argument is not one of the valid sub-commands
+    # or one of the valid optional arguments, then assume that they
+    # are arguments for the "run" sub-command. This allows the
+    # old style command-line invocations to work without modification.
+    if sys.argv[1] not in ['run',
```

Comment:
Nitpick: Maybe the list of available tasks (`run` and `generate`) can defined somewhere else and used here so that if some new task is added in the future, this list will automatically be up to date.

Aside from this (since I see the same block of code in each sub-command's `main` method, is it possible to extract it out to a function that can be used in each?

Reply:
Yeah, I had also noticed the shared code and considered factoring it out but then I reconsidered because the code is quite short, pretty readable, and IMO it's nice to have this bit in the main script itself rather than in another function. For example, it's nice to know that `run_experiment()` is what's called in `rsmtool`, `run_evaluation()` in `rsmeval` etc. 

What do you think?

Reply:
I do like the idea of a list containing the valid subcommands though. Let me do that.

Reply:
What I had envisioned was probably a function that would take a function like `run_experiment` as one of its arguments or maybe something that would sort of prepare it to be called (so you would still see it being used, anyway). I'd have to look into it a bit more. But, nonetheless, I think it's fine the way it is.

Reply:
Yeah, I know what you mean. I tried it after your comment here but it became too much of a meta-mess and not as readable honestly.